docs(README): Use `musl` in `README.md`

razvand opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Building with Newlib results in an error:

```text
 E  /usr/bin/ld: [...]/.unikraft/build/libcontext.o: in function `ukarch_tls_area_init':
 E  [...]/.unikraft/unikraft/arch/x86/x86_64/tls.c:201: undefined reference to `ukarch_tls_tcb_init'
```

There is an issue with Newlib. Until this is fixed, there are two approaches:

i) use Musl
b) remove the libc dependency altogether

Since Musl is the default Unikraft libc, use that. Replace `newlib` with `musl` in the recommended contents for `Kraftfile`.

razvand committed:
docs(README): Remove trailing whitespace

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

razvand committed:
docs(README): Replace newlib with musl

Building with Newlib results in an error:

```
 E  /usr/bin/ld: [...]/.unikraft/build/libcontext.o: in function `ukarch_tls_area_init':
 E  [...]/.unikraft/unikraft/arch/x86/x86_64/tls.c:201: undefined reference to `ukarch_tls_tcb_init'
```

There is an issue with Newlib. Until this is fixed, there are two
approaches:

i) use Musl
b) remove the libc dependency altogether

Since Musl is the default Unikraft libc, use that. Replace `newlib` with
`musl` in the recommended contents for `Kraftfile`.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

craciunoiuc reviewed (approved):
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

razvand committed:
docs(README): Use qemu as platform

Use `qemu` (instead of `kvm`) as platform name in the sample
`Kraftfile` in `README.md`.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

craciunoiuc reviewed (approved):
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
